subprojects/gradle/avito-gradle-api/src/main/kotlin/com/avito/gradle/AvitoTask.kt

```diff
@@ -0,0 +1,16 @@
+package com.avito.gradle
+
+import com.avito.logger.Logger
+import com.avito.logger.LoggerFactory
+import org.gradle.api.DefaultTask
+import org.gradle.api.tasks.Internal
+
+abstract class AvitoTask : DefaultTask() {
+
+    @get:Internal
+    protected val logger: Logger = LoggerFactory.create(
+        className = javaClass.simpleName,
+        project = project,
+        taskName = name
+    )
```

Comment:
I'd like to avoid inheritance if possible.
This logger doesn't depend on the task's state and can be created without inheritance.

Reply:
base class removed